### PR TITLE
refactor: remove duplicate deconstruction in components/form/Form.tsx

### PR DESCRIPTION
**Pull request by wf-joeburton**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [x] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [x] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [ x] Doc is updated/provided or not needed
- [x ] Demo is updated/provided or not needed
- [x ] TypeScript definition is updated/provided or not needed
- [x ] Changelog is provided or not needed

### CODE REVIEW

**Comment by eugene-matvejev** [comment]
👍

File: components/form/Form.tsx [hunk 1]
```diff
@@ -50,6 +48,7 @@ const InternalForm: React.ForwardRefRenderFunction<unknown, FormProps> = (props,
     scrollToFirstError,
     requiredMark,
     onFinishFailed,
+    name,
```
Comment (kerm1it) [inline components/form/Form.tsx:51]: Thanks for your contribution. 

But we need to pass `name` to `rc-filed-form`.

The `restFormProps` will drop `name` if you use destruction.

Comment (hengkx) [reply components/form/Form.tsx:51]: add test?

File: components/form/Form.tsx [hunk 2]
```diff
@@ -118,6 +117,7 @@ const InternalForm: React.ForwardRefRenderFunction<unknown, FormProps> = (props,
         <FieldForm
           id={name}
           {...restFormProps}
+          name={name}
```
Comment (yoyo837) [inline components/form/Form.tsx:120]: @kerm1it 这样对吗？

Comment (kerm1it) [reply components/form/Form.tsx:120]: 可以，但是感觉原来那样就挺好的😂

**Comment by afc163** [comment]
```diff
- - [ x] 
+ - [x] 
```

```diff
- - [x ] 
+ - [x] 
```

**Pull request merged**